lib/engines/content_block_manager/app/models/content_block_manager/content_block/edition.rb

```diff
@@ -58,8 +58,13 @@ def update_object_with_details(object_type, object_title, body)
         details[object_type][object_title] = remove_destroyed body.to_h
       end
 
-      def key_for_object(object)
-        object["title"]&.parameterize.presence || SecureRandom.alphanumeric.downcase
+      def key_for_object(object_type, object)
+        if object["title"].present?
+          object["title"].parameterize
+        else
+          current_count = details[object_type]&.values&.count || 0
+          "#{object_type.parameterize}-#{current_count + 1}"
```

Comment:
just to nit-pick the terminology in the commit message, I think you meant to write "**increment** this count" rather than "prepend this count"